pkg/validation/relabeling_test.go

```diff
@@ -0,0 +1,147 @@
+package validation
+
+import (
+	"bytes"
+	"flag"
+	"testing"
+
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/model/relabel"
+	"github.com/stretchr/testify/require"
+
+	phlaremodel "github.com/grafana/pyroscope/pkg/model"
+)
+
+type wrappedRuntimeConfig struct {
+	rc *RuntimeConfigValues
+}
+
+func (w *wrappedRuntimeConfig) TenantLimits(tenantID string) *Limits {
+	return w.rc.TenantLimits[tenantID]
+}
+
+func (w *wrappedRuntimeConfig) AllByTenantID() map[string]*Limits {
+	return w.rc.TenantLimits
+}
+
+func newOverrides(rc *RuntimeConfigValues) (*Overrides, error) {
+	var defaultCfg Limits
+	fs := flag.NewFlagSet("test", flag.PanicOnError)
+	defaultCfg.RegisterFlags(fs)
+	return NewOverrides(defaultCfg, &wrappedRuntimeConfig{rc})
+}
+
+const tenantOverrideConfig = `
+overrides:
+  nothing: {}
+  disabled:
+    ingestion_relabeling_default_rules_position: disabled
+  custom-rule-end:
+    ingestion_relabeling_rules:
+      - action: drop
+  custom-rule-start:
+    ingestion_relabeling_default_rules_position: last
+    ingestion_relabeling_rules:
+      - action: drop
+  custom-rule-only:
+    ingestion_relabeling_default_rules_position: disabled
+    ingestion_relabeling_rules:
+      - action: drop
+  
```

Comment:
Ah, it will take some time to get used to this notation

Reply:
Yeah, what I meant is overrides within overrides. Although we don't have to use it, a simple plain order will work well for us